src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -60,12 +60,12 @@ public InProcOrRemoteHostAnalyzerRunner(DiagnosticAnalyzerService owner, Abstrac
                     return await AnalyzeInProcAsync(analyzerDriver, project, cancellationToken).ConfigureAwait(false);
                 }
 
-                // due to OpenFileOnly analyzer, we need to run inproc as well for such analyzers for fix all
+                // due to in-process only analyzers, we need to run inproc as well for such analyzers for fix all
                 // otherwise, we don't need to run open file only analyzers for closed files even if full solution analysis is on (perf improvement)
                 //
                 // we have this open file analyzers since some of our built in analyzers such as SimplifyTypeNamesDiagnosticAnalyzer are too
                 // slow to run for whole solution when full solution analysis is on. easily taking more than an hour to run whole solution.
-                var inProcResultTask = AnalyzeInProcAsync(CreateAnalyzerDriver(analyzerDriver, a => forcedAnalysis && a.IsOpenFileOnly(project.Solution.Workspace)), project, remoteHostClient, cancellationToken);
+                var inProcResultTask = AnalyzeInProcAsync(CreateAnalyzerDriver(analyzerDriver, a => (forcedAnalysis || !a.IsOpenFileOnly(workspace)) && a.IsInProcessOnly()), project, remoteHostClient, cancellationToken);
```

Comment:
so. I am fine with bunch of analyzers now returning false for open file only. but I am not sure why we need IsInProcessOnly is needed? 

when we would ever want to run an analyzer for closed files in proc except fix all? (fix all is fine since it is explicitly user requested work) so what open file only: false && in proc: true mean?

also, OOP doesn't have concept of open/close file. so open file only: true && in proc: false doesn't make sense.

Reply:
> open file only: false && in proc: true 

In theory, this is an analyzer that uses services provided by Visual Studio that are not available out-of-process. All of these analyzers should be corrected so they can run out of process.

> OOP doesn't have concept of open/close file. so open file only: true && in proc: false doesn't make sense.

The goal is to eliminate both flags. It will be easier after this separation because the review can focus on the specific parts that it was using instead of trying to review both concepts together.

Reply:
> In theory, this is an analyzer that uses services provided by Visual Studio that are not available out-of-process. All of these analyzers should be corrected so they can run out of process.

this condition is there specifically so that we never run analyzers in proc full solution except fix all.

so, it looks like it opens the door for analyzers to run in proc for whole solution?

Reply:
sure. I think people are okay. I don't see any issue to block this from checking in.